packages/ember-glimmer/tests/integration/components/curly-components-test.js

```diff
@@ -917,4 +920,1044 @@ moduleFor('Components test: curly components', class extends RenderingTest {
+    });
+
+    this.registerComponent('x-middle', {
+      ComponentClass: Component.extend({
+        init() {
+          this._super(...arguments);
+          middle = this;
+        },
+        value: null
+      }),
+      template: '<div id="middle-value">{{value}}</div>{{yield}}'
+    });
+
+    this.registerComponent('x-inner', {
+      ComponentClass: Component.extend({
+        value: null,
+        pushDataUp: Ember.observer('value', function() {
+          middle.set('value', this.get('value'));
+        })
+      }),
+      template: '<div id="inner-value">{{value}}</div>'
+    });
+
+    this.render('{{x-outer}}');
+
+    assert.equal(this.$('#inner-value').text(), '1', 'initial render of inner');
+    assert.equal(this.$('#middle-value').text(), '', 'initial render of middle (observers do not run during init)');
+
+    this.runTask(() => this.rerender());
+
+    assert.equal(this.$('#inner-value').text(), '1', 'initial render of inner');
+    assert.equal(this.$('#middle-value').text(), '', 'initial render of middle (observers do not run during init)');
+
+    this.runTask(() => outer.set('value', 2));
+
+    assert.equal(this.$('#inner-value').text(), '2', 'second render of inner');
+    assert.equal(this.$('#middle-value').text(), '2', 'second render of middle');
+
+    this.runTask(() => outer.set('value', 3));
+
+    assert.equal(this.$('#inner-value').text(), '3', 'third render of inner');
+    assert.equal(this.$('#middle-value').text(), '3', 'third render of middle');
+
+    this.runTask(() => outer.set('value', 1));
+
+    assert.equal(this.$('#inner-value').text(), '1', 'reset render of inner');
+    assert.equal(this.$('#middle-value').text(), '1', 'reset render of middle');
+  }
+
+  ['@test non-block with each rendering child components']() {
+    this.registerComponent('non-block', {
+      template: strip`
+        In layout. {{#each items as |item|}}
+          [{{child-non-block item=item}}]
+        {{/each}}`
+    });
+
+    this.registerComponent('child-non-block', {
+      template: 'Child: {{item}}.'
+    });
+
+    let items = emberA(['Tom', 'Dick', 'Harry']);
+
+    this.render('{{non-block items=items}}', { items });
+
+    this.assertText('In layout. [Child: Tom.][Child: Dick.][Child: Harry.]');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('In layout. [Child: Tom.][Child: Dick.][Child: Harry.]');
+
+    this.runTask(() => this.context.get('items').pushObject('Sergio'));
+
+    this.assertText('In layout. [Child: Tom.][Child: Dick.][Child: Harry.][Child: Sergio.]');
+
+    this.runTask(() => this.context.get('items').shiftObject());
+
+    this.assertText('In layout. [Child: Dick.][Child: Harry.][Child: Sergio.]');
+
+    this.runTask(() => this.context.set('items', emberA(['Tom', 'Dick', 'Harry'])));
+
+    this.assertText('In layout. [Child: Tom.][Child: Dick.][Child: Harry.]');
+  }
+
+  ['@skip specifying classNames results in correct class'](assert) {
```

Comment:
In both `htmlbars` and `glimmer`, specifying a concatenated property in the invocation makes that property not longer _concatenated_ after the rerender. In this case, the component only has the class `baz` after rerendering.